src/ucai/core/utils/callable_utils.py

```diff
@@ -0,0 +1,361 @@
+import ast
+import inspect
+from dataclasses import dataclass
+from textwrap import dedent, indent
+from typing import Any, Callable, Optional, Union, get_args, get_origin, get_type_hints
+
+from ucai.core.utils.type_utils import python_type_to_sql_type
+
+FORBIDDEN_PARAMS = ['self', 'cls']
+
+
+@dataclass
+class DocstringInfo:
+    """Dataclass to store parsed docstring information."""
+    description: str
+    params: Optional[dict[str, str]]
+    returns: Optional[str]
+    raises: Optional[str]
```

Comment:
How is `raises` used? I didn't find it 🤔

Reply:
Removed this as the code that used it is now dead